TestProject/ClientApp/src/components/Posts.js

```diff
@@ -1,11 +1,12 @@
 ﻿import React, { useEffect, useState } from 'react';
 import Post from './Post'
 
-function Posts() {
+function Posts({ userLoggedIn, userProfile, userPosts }) {
     const [posts, setPosts] = useState([]);
 
     useEffect(() => {
-        fetch("/posts/getAllPosts", { method: 'get', mode: 'no-cors' })
+        const url = userPosts ? `/posts/getPosts/${userProfile.id}` : "/posts/getPosts";
```

Comment:
you don't need 2 separate API methods to retrieve the same info. Pass options and filter posts by them on the backend.

Reply:
In general it's a bad approach use the same request for different calls.